mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessage.java

```diff
@@ -49,7 +49,7 @@
 public class IndexableMessage {
 
     public static IndexableMessage from(MailboxMessage message, List<User> users, TextExtractor textExtractor, 
-            ZoneId zoneId, IndexAttachments indexAttachments) {
+            ZoneId zoneId, IndexAttachments indexAttachments, boolean keepMessageId) {
```

Comment:
I don't agree with this changeset : you actually use a single type for two purposes.
A result should not depend on the system capability : either it works or it doesn't;